### PR TITLE
DAOS-5217 test: Switch all CI IOR tests to use the DFS driver instead of DAOS

### PR DESCRIPTION
**Pull request by mchaarawi**
- DAOS driver is deprecated and will be removed soon.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/tests/ftest/io/io_aggregation.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ container:
   type: POSIX
   control_method: daos
 ior:
-  api: "DAOS"
+  api: "DFS"
   client_processes:
     np: 1
   daos_destroy: False
```
Comment (saurabhtandan) [inline src/tests/ftest/io/io_aggregation.yaml:38]: You need to change this to `dfs_destroy`

[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/tests/ftest/io/io_aggregation.yaml]: fixed

File: src/tests/ftest/util/ior_utils.py [hunk 2]
```diff
@@ -286,17 +254,11 @@ def get_default_env(self, manager_cmd, log_file=None):
...
-                env["DAOS_CONT"] = self.daos_cont.value
-                env["IOR_HINT__MPI__romio_daos_obj_class"] = \
-                    self.daos_oclass.value
-            elif self.dfs_pool.value is not None:
+            if self.dfs_pool.value is not None:
                 env["DAOS_POOL"] = self.dfs_pool.value
                 env["DAOS_SVCL"] = self.dfs_svcl.value
                 env["DAOS_CONT"] = self.dfs_cont.value
-                env["IOR_HINT__MPI__romio_dfs_obj_class"] = \
+                env["IOR_HINT__MPI__romio_daos_obj_class"] = \
```
Comment (sylviachanoiyee) [inline src/tests/ftest/util/ior_utils.py:261]: Should this be *_dfs_obj_class ?

Comment (mchaarawi) [reply src/tests/ftest/util/ior_utils.py:261]: nope, that's the MPI-IO hint. it's daos

Comment (sylviachanoiyee) [reply src/tests/ftest/util/ior_utils.py:261]: ok thanks

**Comment by mchaarawi** [comment]
All failures are unrelated to IOR (IOR changes have passed).
Master has these failures already:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/lastFailedBuild/#showFailuresLink

**Pull request merged**